lib/gather/collector.rb

```diff
@@ -40,18 +41,33 @@ def self.physical_data
       data[:model] = `dmidecode -s system-product-name`.chomp
       data[:bios] = `dmidecode -s bios-version`.chomp
       data[:serial] = `dmidecode -s system-serial-number`.chomp
-      data[:ram] = `grep MemTotal /proc/meminfo | awk '{print $2}'`.chomp # RAM measured in kB
+
+      # Get RAM info
+
+      data[:ram] = {}
+      ram_info = `dmidecode -q -t 17`.split('Memory Device')[1..]
+      data[:ram][:total_capacity] = si_to_bytes(between(`dmidecode -t 16`, 'Maximum Capacity: ', "\n")) / (1000.0 ** 3)
```

Comment:
This is getting the total possible memory that the system could support, not the amount it currently has. 

The current total memory can be found with, for example, the lshw command:
```
lshw -quiet -c memory 2>/dev/null |grep size |grep -o '[[:digit:]]*'
```

It could also be found by getting the capacity of modules from the `dmidecode` output and summing them together.

Reply:
Fixed edad9c1